docs/modules/sql/pages/user-defined-types.adoc

```diff
@@ -1,60 +1,53 @@
-more complex cases of fully/partially connected graphs of objects. See <<queryingExamples,examples>> for more information.
-
-WARNING: Support for UDTs is an experimental feature, and disabled by default.
-See <<enabling-udt-support, Enabling UDT support>> to learn how to enable it.
+UDTs are useful for creating and accessing hierarchical data structures, which can even be cyclic. See <<queryingExamples,examples>> for more information.
 
 == Feature Overview
 
-Due to the experimental nature of UDTs, the feature set is limited to the following:
+- All serialization formats (namely `portable`, `compact`, `java` and `avro`) are supported.
+- Cyclic definitions are only supported for Java-serialized structures. Portable and Compact formats do not support cyclic schemas, and Avro support is currently limited to acyclic schemas.
++
+NOTE: If mappings use cyclic structures, the `INSERT` and `UPDATE` queries are disabled.
+- Type options override mapping options. Where both the mapping and type define a schema (Portable class definition, Compact schema, Java class or Avro schema) and the mapping contains only `__key` or `this` fields, the schema defined by the type is used. This means that the mapping does not need to define a schema if the type defines one.
+- If the type does not specify a schema, the parent structure is used. The parent structure could be a mapping or another type.
+- Type fields are optional and are resolved from the schema if not specified. Following a `CREATE MAPPING` statement, type fields are resolved and materialized. You can check the resolved fields using a `GET_DDL` query after the type has been used in a mapping. If type fields are not specified:
 
-- Support for three formats - `portable`, `compact` and `java` with varying level of sub-feature support
-- Instance and type-level cycles are only supported for Java types.
-- Type formats (referred to as type kind or kind hereafter) can not be mixed, both between mappings and types themselves.
-For example, you cannot use Java type in another Portable type or Portable mapping.
-- Limited support for instance and type-level cycles; only for Java types.
-Note that INSERT/UPDATE are disabled for mappings that use type hierarchies that contain cycles.
-- Limited support for `INSERT` and `UPDATE` queries for mappings using UDTs; if a `TYPE` hierarchy contains cycles,
-`INSERT` and `UPDATE` statements are disabled for the mapping.
-- Support for using UDT-based projections in both normal `SELECT` projection lists, `WHERE` filters and in `JOIN` conditions.
+ ** For Portable, Compact, or Avro formats, schema fields cannot be complex (`PORTABLE`, `COMPACT`, or `RECORD`
+ ). 
+ ** For Java format, complex fields are allowed and map to `OBJECT`. 
 
-== Enabling UDT Support
-You can enable UDTs by setting the `hazelcast.sql.experimental.custom.types.enabled` property to `true` in the member configuration.
+== Enabling Cycling Type Support
```

Comment:
I would move this section toward end of the document

Reply:
It is now mentioned in "Support for Cycles". (099f756d)